godtools/App/Flows/OnboardingFlow.swift

```diff
@@ -43,16 +43,39 @@ class OnboardingFlow: Flow {
         let viewModel = OnboardingTutorialViewModel(
             flowDelegate: self,
             analyticsContainer: appDiContainer.analytics,
+            tutorialVideoAnalytics: appDiContainer.getTutorialVideoAnalytics(),
             onboardingTutorialItemsRepository: onboardingTutorialItemsRepository,
             onboardingTutorialAvailability: appDiContainer.onboardingTutorialAvailability,
             openTutorialCalloutCache: appDiContainer.openTutorialCalloutCache,
-            customViewBuilder: appDiContainer.onboardingTutorialCustomViewBuilder(),
+            customViewBuilder: appDiContainer.onboardingTutorialCustomViewBuilder(flowDelegate: self),
             localizationServices: appDiContainer.localizationServices
         )
         let view = OnboardingTutorialView(viewModel: viewModel)
         
         navigationController.setViewControllers([view], animated: false)
     }
+    
+    private func launchVideoPlayerView(youtubeVideoId: String) {
+        
+        let viewModel = VideoPlayerViewModel(flowDelegate: self, youtubeVideoId: youtubeVideoId, closeVideoPlayerFlowStep: .closeVideoPlayerTappedFromOnboardingTutorial, videoEndedFlowStep: .videoEndedOnOnboardingTutorial)
+        
+        let view = VideoPlayerView(viewModel: viewModel)
+        
+        let modal = ModalNavigationController(rootView: view)
+        modal.navigationBar.setupNavigationBarAppearance(backgroundColor: UIColor.black, controlColor: nil, titleFont: nil, titleColor: nil, isTranslucent: true)
```

Comment:
Hey @reldredge71 a couple things here.
1. This will get called before ```viewDidLoad``` of the ModalNavigationController, this doesn't appear to cause any issues now, but it potentially could.
2.  Instead, I'd like ModalNavigationController to make use of the new ```setupNavigationBarAppearance``` in place of the current logic.  We just need to ensure this change will be transparent to any other places in the code base where ModalNavigationController is used by keeping the defaults white and opaque navigation bar.  

My thinking would be to provide arguments on initialization that are set to default values, this way any current places using ModalNavigationController will not have to make any changes.

```
import UIKit

class ModalNavigationController: UINavigationController {
    
    private static let defaultNavBarColor: UIColor = .white
    private static let defaultNavBarIsTranslucent: Bool = false
    
    private let rootView: UIViewController
    private let navBarColor: UIColor
    private let navBarIsTranslucent: Bool
    
    required init(rootView: UIViewController, navBarColor: UIColor = ModalNavigationController.defaultNavBarColor, navBarIsTranslucent: Bool = ModalNavigationController.defaultNavBarIsTranslucent) {
        
        self.rootView = rootView
        self.navBarColor = navBarColor
        self.navBarIsTranslucent = navBarIsTranslucent
        
        super.init(nibName: nil, bundle: nil)
       
        modalPresentationStyle = .fullScreen
    }
    
    ...
}

...

```